pkg/kubectl/cmd/delete_test.go

```diff
@@ -70,7 +70,7 @@ func TestDeleteNamedObject(t *testing.T) {
 		Codec: codec,
 		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
 			switch p, m := req.URL.Path, req.Method; {
-			case p == "/namespaces/test/replicationcontrollers/redis-master-controller" && m == "DELETE":
+			case p == "/namespaces/test/replicationcontrollers/redis-master-controller" && (m == "DELETE" || m == "GET"):
 				return &http.Response{StatusCode: 200, Body: objBody(codec, &rc.Items[0])}, nil
 			default:
 				// Ensures no GET is performed when deleting by name
```

Comment:
so I think we need the following tests:
* GET succeeds, and is followed by DELETE
* GET returns 404... not sure if DELETE should be performed or not
* GET returns 403, and is followed by DELETE, which succeeds